### DISCUSSION TITLE
Connection timeout

### DISCUSSION THREAD

**Discussion by whgibbo** [psycopg/psycopg2#1489] [General]
Hi,
New to using psycopg2.. Looking thru the documentation I could see a way to check/change the connection timeout.

Any thoughts?
Thanks

**Comment by dvarrazzo** [comment]
You can pass the `connect_timeout` as a keyword argument or DSN string option. See [the libpq docs](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS) for a list of all the parameters supported.

Note that it doesn't work for async connections (including greenlet connections). The problem is solved in psycopg 3, but not backported.
